src/adapters/RoomsJSONAdapter.js

```diff
@@ -74,9 +76,32 @@ export default class RoomsJSONAdapter extends RoomsAdapter {
     return Observable.create((observer) => {
       const data = !this.datasource[`${ID}-activities`] ? [] : this.datasource[`${ID}-activities`];
 
-      observer.next(data);
+      const formattedData = RoomsJSONAdapter.formatActivity(data);
+
+      observer.next(formattedData);
 
       observer.complete();
     });
   }
 }
+
+/**
+ * Activities could be objects with metadata (like date separators).
+ * Convert the data into proper javascript objects
+ *
+ * @param {Array} activityData json formatted activity data
+ * @returns {Array}
+ */
+RoomsJSONAdapter.formatActivity = (activityData) =>
```

Comment:
can we implement the function the way we did it in all other adapters?

Reply:
I'm not sure I understand what you are asking

Reply:
Lalli explained it better [here](https://github.com/webex/components/pull/34#discussion_r325929812)